[AArch64] Update feature dep. for Armv9.6 extensions

CarolineConcatto opened the pull request:
These features FEAT_FAMINMAX, FEAT_LUT and FEAT_FP8 depends on FEAT_NEON.

Update dependency from FEAT_FP8DOT4 and FEAT_FP8DOT2. Now depends indirectly on FEAT_NEON through FEAT_FP8

CarolineConcatto committed:
[AArch64] Update feature dep. for Armv9.6 extensions

These features FEAT_FAMINMAX, FEAT_LUT and FEAT_FP8 depends on
FEAT_NEON.

Update dependency from FEAT_FP8DOT4 and FEAT_FP8DOT2. Now depends indirectly
on FEAT_NEON through FEAT_FP8

mgabka commented on llvm/unittests/TargetParser/TargetParserTest.cpp:
```diff
@@ -1978,6 +1988,7 @@ AArch64ExtensionDependenciesBaseArchTestParams
         {AArch64::ARMV9_6A, {"fp8dot4", "nofp8"}, {}, {"fp8", "fp8dot4"}},
         {AArch64::ARMV9_6A, {"nofp8", "fp8dot2"}, {"fp8", "fp8dot2"}, {}},
         {AArch64::ARMV9_6A, {"fp8dot2", "nofp8"}, {}, {"fp8", "fp8dot2"}},
+
```
nit: not needed change

paulwalker-arm commented on llvm/unittests/TargetParser/TargetParserTest.cpp:
```diff
@@ -1828,10 +1828,20 @@ AArch64ExtensionDependenciesBaseArchTestParams
+        {AArch64::ARMV9_6A, {"fp8dot2", "nofp8"}, {}, {"fp8", "fp8dot2"}},
+
+        // simd -> {faminman, lut, fp8}}
+        {AArch64::ARMV9_6A, {"faminmax", "nosimd"}, {}, {"neon", "faminmax"}},
+        {AArch64::ARMV9_6A, {"nosimd", "faminmax"}, {"neon", "faminmax"}, {}},
+        {AArch64::ARMV9_6A, {"lut", "nosimd"}, {}, {"neon", "lut"}},
+        {AArch64::ARMV9_6A, {"nosimd", "lut"}, {"neon", "lut"}, {}},
+        {AArch64::ARMV9_6A, {"fp8", "nosimd"}, {}, {"neon", "fp8"}},
+        {AArch64::ARMV9_6A, {"nosimd", "fp8"}, {"neon", "fp8"}, {}},
```
Can you move these to the `// simd ->` block above so they sit with the other `AArch64::ARMV9_6A` features that have the same dependency.

CarolineConcatto committed:
Address review comments

CarolineConcatto committed:
Fix aarch64-fujitsu-monaka extension test

Explicitly add FEAT_LUT and FEAT_FAMINMAX to the target.
Before they were implicitly inferred through FEAT_FP8.

CarolineConcatto commented:
/cherry-pick [cc0d4a5](https://github.com/llvm/llvm-project/commit/cc0d4a5188f9e41dd8e00048d17b87529e28b694)

CarolineConcatto commented:
/cherry-pick [cc0d4a5](https://github.com/llvm/llvm-project/commit/cc0d4a5188f9e41dd8e00048d17b87529e28b694)

tstellar commented:
Backported as: de5dcad3a2d332e75ea3b680d7d41080b064ca7c
